shared/test/test_sources.rb

```diff
@@ -0,0 +1,49 @@
+require 'minitest/autorun'
+require 'rack/test'
+require File.expand_path '../../../deployment', __FILE__
+require File.expand_path '../../middleware/assets_api', __FILE__
```

Comment:
The file is still called `assets_api.rb` but it has logic for both the sources and assets.  I'll rename this to `files_api.rb` but I thought it would be easier to review with the diff first.